packages/blade/src/components/Chip/Chip.tsx

```diff
@@ -0,0 +1,237 @@
+      if (_isDisabled) return;
+      if (e.key === ' ') {
+        setIsPressed(false);
+      }
+    },
+    [_isDisabled],
+  );
+
+  const chipTextColor =
+    chipColorTokens.text[_isDisabled ? 'disabled' : _isChecked && _intent ? _intent : 'unchecked'];
+  const chipBackgroundColor =
+    chipColorTokens.background[_isChecked && _intent ? _intent : 'unchecked'][
+      _isDisabled ? 'disabled' : 'default'
+    ];
+  const chipBorderColor =
+    chipColorTokens.border[_isChecked && _intent ? _intent : 'unchecked'][
+      _isDisabled ? 'disabled' : 'default'
+    ];
+
+  return (
+    <BaseBox
+      {...metaAttribute({ name: MetaConstants.Chip, testID })}
+      {...getStyledProps(styledProps)}
+      display={(isReactNative() ? 'flex' : 'inline-flex') as never}
+    >
+      <SelectorLabel
+        componentName={MetaConstants.ChipLabel}
+        onTouchStart={handlePointerPressedIn}
+        onTouchEnd={handlePointerPressedOut}
+        onMouseDown={handlePointerPressedIn}
+        onMouseUp={handlePointerPressedOut}
+        onMouseOut={handlePointerPressedOut}
+        onKeyDown={handleKeyboardPressedIn}
+        onKeyUp={handleKeyboardPressedOut}
+        inputProps={isReactNative() ? inputProps : {}}
+        style={{ cursor: _isDisabled ? 'not-allowed' : 'pointer' }}
+      >
+        <BaseBox display="flex" flexDirection="column">
+          <BaseBox display="flex" alignItems="center" flexDirection="row">
+            <SelectorInput
+              hoverTokens={getChipHoverTokens(_intent)}
+              isChecked={state.isChecked}
+              isDisabled={_isDisabled}
+              inputProps={inputProps}
+              ref={ref}
+            />
+            <AnimatedChip
+              borderColor={chipBorderColor as never}
+              isDisabled={_isDisabled}
+              isPressed={isPressed}
+              isDesktop={matchedDeviceType === 'desktop'}
+            >
+              <BaseBox
+                display="flex"
+                flexDirection="row"
+                justifyContent="center"
+                alignItems="center"
+                overflow="hidden"
+                backgroundColor={chipBackgroundColor as never}
+                borderRadius="max"
+                borderWidth={['xsmall', 'small'].includes(_size) ? 'thinner' : 'thin'}
+                paddingLeft={
+                  chipHorizontalPaddingTokens[Boolean(Icon) ? 'icon' : 'default'].left[_size]
+                }
+                paddingRight={
+                  chipHorizontalPaddingTokens[Boolean(Icon) ? 'icon' : 'default'].right[_size]
+                }
+                style={{
+                  height: chipHeightTokens[_size],
+                  borderColor: _isChecked ? 'inherit' : 'transparent',
+                }}
+              >
+                {Icon ? (
+                  <BaseBox paddingRight="spacing.3" display="flex">
+                    <Icon color={chipTextColor as never} size={chipIconSizes[_size]} />
```

Comment:
same here. 

If the TS is throwing error that means we need to change that `ColorTokens` type in chipTokens. 

I seems like currently it's way to big, it includes literally all the color tokens: 
<img width="708" alt="image" src="https://github.com/razorpay/blade/assets/35374649/a6eb2a54-4ab6-456b-9fe1-fed2ba9f104f">

Reply:
Fixed types

Reply:
@anuraghazra I updated it to use `IconColor` type, but it doesn't accept `brand.primary.500` yet used, should I add it to the Icon props?

Reply:
I just confirmed with Abhishek, that these should be text color only. Should I update the type of IconProps?

Reply:
@anuraghazra We can do it in a non-breaking way `IconColorType = currentType | TextProps['color']`

Reply:
we already have this type supported right? `TextIconColors` for icons?

Reply:
@kamleshchandnani TextIconColors are `surface.text.*` tokens. We need `brand.primary.500` and `feedback.text.<intent>.lowContrast`